src/Common/NamePrompter.h

```diff
@@ -90,17 +90,16 @@ class NamePrompter
     }
 };
 
-template <size_t MaxNumHints, class Self>
+
+template <size_t MaxNumHints, typename Self>
 class IHints
 {
 public:
-
     virtual std::vector<String> getAllRegisteredNames() const = 0;
 
     std::vector<String> getHints(const String & name) const
     {
-        static const auto registered_names = getAllRegisteredNames();
```

Comment:
It was ok for singletons like FunctionFactory, DataTypeFactory, etc. but not Ok for Enums. It also not ok after we will introduce UDFs and UDTs.